Framework/Geometry/inc/MantidGeometry/Instrument/DetectorInfoItem.h

```diff
@@ -53,6 +53,12 @@ template <typename T> class DetectorInfoItem {
     return m_detectorInfo->rotation(m_index);
   }
 
+  double l2() const { return m_detectorInfo->l2(m_index); }
+
+  size_t infoSize() const { return m_detectorInfo->size(); }
```

Comment:
This is a very unusual method to have on an iterator, reflected in the unclear name. Is it really needed?

Reply:
Agreed. I can probably get around this.